idaes/core/scaling/tests/test_scaling_profiler.py

```diff
@@ -0,0 +1,554 @@
+        # Dummy method to apply scaling by block
+        def dummy_scaler(block):
+            set_scaling_factor(block.c1, 2)
+            set_scaling_factor(block.c2, 4)
+
+        sp = ScalingProfiler(
+            build_model=demo_model,
+            user_scaling=demo_scaling,
+        )
+
+        # Block based method, so use block_based=True
+        model = sp._build_model()
+        sp._apply_scaling(model, dummy_scaler, block_based=True)
+
+        # Should have scaling factors of for all constraints
+        assert len(model.scaling_factor) == 2
+        assert model.scaling_factor[model.c1] == 2
+        assert model.scaling_factor[model.c2] == 4
+
+    @pytest.mark.unit
+    def test_solve_perturbed_state_no_callback(self):
+        sp = ScalingProfiler(
+            build_model=demo_model,
+            user_scaling=demo_scaling,
+        )
+
+        model = sp._build_model()
+        res = sp._solved_perturbed_state(model)
+
+        assert res == {}
+
+    @pytest.mark.unit
+    def test_solve_perturbed_state(self):
+        sp = ScalingProfiler(
+            build_model=demo_model,
+            user_scaling=demo_scaling,
+            perturb_state=demo_pertubration,
+        )
+
+        model = sp._build_model()
+        res = sp._solved_perturbed_state(model)
+
+        assert value(model.v3) == pytest.approx(-12, rel=1e-6)
+
+        # Model is trivially presolvable, so expect convergence in 0 iterations
+        assert res["solved"]
+        assert (
+            res["termination_message"]
+            == "TerminationCondition.convergenceCriteriaSatisfied"
+        )
+        assert res["iterations"] == 0
+
+
+# Case study using Gibbs reactor model
+# Get solution json from scaling tests
+FILENAME = "gibbs_solution.json"
+local_path = os.path.dirname(os.path.realpath(__file__))
+fname = os.path.join(local_path, FILENAME)
+
+
+def build_model():
+    model = ConcreteModel()
+    model.fs = FlowsheetBlock(dynamic=False)
+
+    model.fs.properties = MethaneCombustionParameterBlock()
+
+    model.fs.unit = GibbsReactor(
+        property_package=model.fs.properties,
+        has_heat_transfer=True,
+        has_pressure_change=True,
+    )
+
+    model.fs.unit.inlet.flow_mol[0].fix(230.0)
+    model.fs.unit.inlet.mole_frac_comp[0, "H2"].fix(0.0435)
+    model.fs.unit.inlet.mole_frac_comp[0, "N2"].fix(0.6522)
+    model.fs.unit.inlet.mole_frac_comp[0, "O2"].fix(0.1739)
+    model.fs.unit.inlet.mole_frac_comp[0, "CO2"].fix(1e-5)
+    model.fs.unit.inlet.mole_frac_comp[0, "CH4"].fix(0.1304)
+    model.fs.unit.inlet.mole_frac_comp[0, "CO"].fix(1e-5)
+    model.fs.unit.inlet.mole_frac_comp[0, "H2O"].fix(1e-5)
+    model.fs.unit.inlet.mole_frac_comp[0, "NH3"].fix(1e-5)
+    model.fs.unit.inlet.temperature[0].fix(1500.0)
+    model.fs.unit.inlet.pressure[0].fix(101325.0)
+
+    model.fs.unit.outlet.temperature[0].fix(2844.38)
+    model.fs.unit.deltaP.fix(0)
+
+    from_json(model, fname=fname, wts=StoreSpec.value())
```

Comment:
What is the purpose of this json file? If I wanted to use the scaling profiler on a flowsheet, would I need to generate a similar json?

Reply:
That is for test purposes so that we can load a solution into the model without having to spend time on initializing it first. You could replace that with a standard initialization and solve, but ti would take longer to run.